telegram/utils/helpers.py

```diff
@@ -113,3 +120,45 @@ def mention_markdown(user_id, name):
     """
     if isinstance(user_id, int):
         return '[{}](tg://user?id={})'.format(escape_markdown(name), user_id)
+
+
+def _extract_urls_from_text(text):
+    """
+    Returns a list of urls from a text string.
+    URLs without a leading `http://` or `www.` won't be found.
+    """
+    out = []
+    for word in text.split(' '):
+        thing = urlparse(word.strip())
+        if thing.scheme:
+            out.append(word)
+    return out
+
+
+def extract_urls(message):
+    """
+    Extracts all Hyperlinks that are contained in a message. This includes
+    message entities and the media caption. Distinct links are returned in order of appearance.
```

Comment:
Please clarify that message links will be always be specified before caption links.

Reply:
that actually makes no sense, because a message has either text or a caption, never both